File: js/complete/no016.js
#!/usr/bin/env node

/* What is the sum of the digits of the number 2^(1000)? */

var bigint = require('bigint'),
    operator = require('../operator.js'),
    timer = require('../timer.js');

exports.main = function(verbose) {
    if (typeof verbose == 'undefined') {
        verbose = false;
    }
    var power = bigint('2').pow(1000).toString(),
        digitArray = power.split('');
    return operator.sum(digitArray);
};

if (require.main === module) {
    timer.timer(16, exports.main);
}

File: python/complete/no016.py
#!/usr/bin/env python

# What is the sum of the digits of the number 2^(1000)?

from python.decorators import euler_timer

def main(verbose=False):
    return sum([int(dig) for dig in str(2**1000)])

if __name__ == '__main__':
    print euler_timer(16)(main)(verbose=True)
